serverless/serverless
# Credential Fixes Roundup

## What did you implement:

Merge fixes from both #2373 & #2695, adding editorial suggestions and future-proofing the code a bit (probably)



## How did you implement it:

Merge the great work of @stevecaldwell77 and @rkistner and apply the organizational pattern that had been in place to those changes.

Clean up tests to broadly clean and restore process/test environment and simplify test code.

This work uncovered that we were returning profile values in the profile attribute when no profile credentials were loaded.  This seems inappropriate.  The tests were modified to use the temporary credential file creation code that @stevecaldwell77 wrote exclusively.  This way there are actual credentials to load from the given profile.  Previously it was accepted that we couldn't guarantee an existing profile with our reserved names and values for all contributors.  It seemed this would create a barrier to contribution so it was accepted.  Using the temporary credentials file improves that circumstance although we won't be testing loading from the default location.  I think the responsibility for that working that responsibility rests with the aws-sdk.



## How can we verify it:

I am not sure how to test the assume-role/asynchronous STS credential loading capability and would love some help making sure that is right.  Both of the linked & merged PRs have some instructions on how to verify that capability.

You can set the `AWS_SHARED_CREDENTIALS_FILE` environment variable to test loading credentials from an arbitrary location.  You might rename your `~/.aws/credentials` file and reference the renamed file for example.  Don't forget to rename it back for convenience.




## Todos:



***Is this ready for review?:*** YES
